src/engine/TransitivePath.cpp

```diff
@@ -215,361 +218,168 @@ size_t TransitivePath::getCostEstimate() {
-    // the max step limit
-    while (!positions.empty()) {
-      size_t stackIndex = positions.size() - 1;
-      // Process the next child of the node at the top of the stack
-      ad_utility::HashSet<Id>::const_iterator& pos = positions[stackIndex];
-      const ad_utility::HashSet<Id>* nodeEdges = edgeCache.back().get();
+  *dynRes = std::move(res).toDynamic();
+}
 
-      if (pos == nodeEdges->end()) {
-        // We finished processing this node
-        positions.pop_back();
-        edgeCache.pop_back();
-        continue;
-      }
+// _____________________________________________________________________________
+ResultTable TransitivePath::computeResult() {
+  LOG(DEBUG) << "TransitivePath subresult computation..." << std::endl;
+  shared_ptr<const ResultTable> subRes = _subtree->getResult();
+  LOG(DEBUG) << "TransitivePath subresult computation done." << std::endl;
 
-      Id child = *pos;
-      ++pos;
-      size_t childDepth = positions.size();
-      if (childDepth <= maxDist && marks.count(child) == 0) {
-        // process the child
-        if (childDepth >= minDist) {
-          marks.insert(child);
-          if constexpr (rightIsVar) {
-            res.push_back({nodes[i], child});
-          } else {
-            res.push_back({child, nodes[i]});
-          }
-        }
-        // Add the child to the stack
-        MapIt it = edges.find(child);
-        if (it != edges.end()) {
-          positions.push_back(it->second->begin());
-          edgeCache.push_back(it->second);
-        }
-      }
-    }
+  IdTable idTable{getExecutionContext()->getAllocator()};
 
-    if (i + 1 < nodes.size()) {
-      // reset everything for the next iteration
-      marks.clear();
-    }
-  }
+  idTable.setNumColumns(getResultWidth());
 
-  *dynRes = std::move(res).toDynamic();
-}
+  size_t subWidth = subRes->idTable().numColumns();
+  if (_lhs.treeAndCol.has_value()) {
+    LOG(DEBUG) << "TransitivePath left result computation..." << std::endl;
+    shared_ptr<const ResultTable> leftRes =
+        _lhs.treeAndCol.value().first->getResult();
+    LOG(DEBUG) << "TransitivePath left result computation done." << std::endl;
+    size_t leftWidth = leftRes->idTable().numColumns();
 
-template <size_t SUB_WIDTH, size_t LEFT_WIDTH, size_t RES_WIDTH>
-void TransitivePath::computeTransitivePathLeftBound(
-    IdTable* dynRes, const IdTable& dynSub, const IdTable& dynLeft,
-    size_t leftSideCol, bool rightIsVar, size_t leftSubCol, size_t rightSubCol,
-    Id rightValue, size_t minDist, size_t maxDist, size_t resWidth) {
-  using Map = ad_utility::HashMap<Id, std::shared_ptr<ad_utility::HashSet<Id>>>;
-  using MapIt = Map::iterator;
+    LOG(DEBUG) << "TransitivePath result computation..." << std::endl;
+    CALL_FIXED_SIZE((std::array{_resultWidth, subWidth, leftWidth}),
+                    &TransitivePath::computeTransitivePathBound, this, &idTable,
+                    subRes->idTable(), _lhs, _rhs, leftRes->idTable());
```

Comment:
We don't really need the `DEBUG` statements (we have better means for this now), 
And the `left` and `right` case is very similar and can be factored out to a lambda or function.